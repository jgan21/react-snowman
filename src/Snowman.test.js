import React from "react";
import { render, fireEvent, debug } from "@testing-library/react";
import Snowman from "./Snowman";
import img0 from "./0.png";
import img1 from "./1.png";
import img2 from "./2.png";
import img3 from "./3.png";
import img4 from "./4.png";
import img5 from "./5.png";
import img6 from "./6.png";


function clickButton(container, ltr) {
  return fireEvent.click(container.querySelector(`button[value="${ltr}"]`));

}

// make sure the image is there
// make sure the buttons are gone
// make sure the message shows = "You lose" and the correct word
describe("works: game stops at max guesses", function () {
  test("should show image after max wrong guesses", function () {
    const { container } = render(<Snowman images={[img0, img1, img2, img3, img4, img5, img6]}
      words={["apple"]}
      maxWrong={6} />);

    clickButton(container, "k");
    clickButton(container, "m");
    clickButton(container, "n");
    clickButton(container, "o");
    clickButton(container, "d");
    clickButton(container, "c");
    // clickButton(container, "q");

    expect(container).toContainHTML("<img src");

    const img = container.querySelector("img");
    expect(img.getAttribute("src"))
      .toEqual("6.png");

  });

  test("makes sure the buttons are gone after max wrong guesses", function () {
    const { container } = render(<Snowman images={[img0, img1, img2, img3, img4, img5, img6]}
      words={["apple"]}
      maxWrong={6} />);

    clickButton(container, "k");
    clickButton(container, "m");
    clickButton(container, "n");
    clickButton(container, "o");
    clickButton(container, "d");
    clickButton(container, "c");

    expect(container.querySelector("button")).toBeNull();

  });

  test("makes sure the message You lose shows after max wrong guesses", function () {
    const { container } = render(<Snowman images={[img0, img1, img2, img3, img4, img5, img6]}
      words={["apple"]}
      maxWrong={6} />);

    clickButton(container, "k");
    clickButton(container, "m");
    clickButton(container, "n");
    clickButton(container, "o");
    clickButton(container, "d");
    clickButton(container, "c");

    expect(container).toContainHTML("You lose");
    expect(container.querySelector(".Snowman-word")).toContainHTML("apple");

  });



});

describe("snapshot", function () {
  test("matches snapshot", function () {
    const { container } = render(<Snowman images={[img0, img1, img2, img3, img4, img5, img6]}
      words={["apple"]}
      maxWrong={6} />);

    clickButton(container, "k");
    clickButton(container, "m");
    clickButton(container, "n");
    clickButton(container, "o");
    clickButton(container, "d");
    clickButton(container, "c");

    expect(container).toMatchSnapshot();
  });
});