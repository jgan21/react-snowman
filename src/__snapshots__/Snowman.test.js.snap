// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p
      class="Snowman-word-count"
    >
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      apple
    </p>
    <p>
      You lose!
    </p>
    <button
      class="Snowman-restart"
    >
      Restart
    </button>
  </div>
</div>
`;
